### PR TITLE
COR-523 Implement sorting

### PR DESCRIPTION
**Pull request by ludydoo**

### CODE REVIEW

**Comment by ludydoo** [comment]
There was a missing closing `<div>` in the `individuals.gohtml` hence the large diff

File: internal/db/individual.go [hunk 1]
```diff
@@ -392,3 +419,28 @@ func (i individualRepo) softDeleteManyInternal(ctx context.Context, tx *sqlx.Tx,
...
+		return constants.DBColumnIndividualFullName, nil
+	case "phoneNumber":
+		return constants.DBColumnIndividualPhoneNumber, nil
+	case "normalizedPhoneNumber":
+		return constants.DBColumnIndividualNormalizedPhoneNumber, nil
+	case "age":
+		return constants.DBColumnIndividualBirthDate, nil
+	case "gender":
+		return constants.DBColumnIndividualGender, nil
+	case "email":
```
Comment (neb42) [inline internal/db/individual.go:441]: Don't we have constants we can use for these case strings?

File: internal/db/individual.go [hunk 2]
```diff
@@ -192,7 +193,33 @@ func (i individualRepo) unbatchedGetAllInternal(ctx context.Context, tx *sqlx.Tx
...
+			case api.SortDirectionAscending:
+				query = query + " ASC"
+			case api.SortDirectionDescending:
+				query = query + " DESC"
+			default:
+				query = query + " ASC"
+			}
+		}
+	} else {
+		query = query + " ORDER BY created_at DESC"
```
Comment (neb42) [inline internal/db/individual.go:220]: I think there was a product requirement to order by id, but I think `created_at` is better. Maybe just something to raise with Fix

File: web/templates/individuals.gohtml [hunk 3]
```diff
@@ -323,120 +412,172 @@
...
+    {{$maxWidth := .MaxWidth}}
+    {{$sortDirection := $options.GetSortDirectionForField $fieldName}}
+    {{$ascSortUrl := $options.GetSortURLForField $fieldName "ascending"}}
+    {{$descSortUrl := $options.GetSortURLForField $fieldName "descending"}}
+
+    {{ if not $sortable}}
+        <th style="min-width: 5rem">
+            <a class="d-inline-block text-decoration-none text-black">
+                {{$label}}
+            </a>
```
Comment (neb42) [inline web/templates/individuals.gohtml:549]: Why an anchor tag in the column header?

**Review by neb42 — COMMENTED** [review]
On hover the sort icon should be orange

File: internal/db/individual.go (hunk 1 above)
Comment (ludydoo) [reply internal/db/individual.go:441]: Yeah I've set up a pr to generate the `json` tag constants, we could use that

File: internal/db/individual.go (hunk 2 above)
Comment (ludydoo) [reply internal/db/individual.go:220]: Because sorting by UUID.. does not make much sense

File: web/templates/individuals.gohtml (hunk 3 above)
Comment (ludydoo) [reply web/templates/individuals.gohtml:549]: whoops. Corrected

**Comment by neb42** [comment]
> On hover the sort icon should be orange

Still needs this

**Pull request closed without merging**